src/LinkedList.ts

```diff
@@ -2,54 +2,52 @@ import { fabric } from "fabric";
 import { Node } from "./Node";
 import { makeLine } from "./Utils";
 import Config = require("./Config");
+import { Variable } from "./Variable";
 
 export class LinkedList {
 
-    private head: Node = null;
+    private globalVars: { [key: string]: Variable };
```

Comment:
I had it in my head that this should be "DSPlayground" object so that it can be reused when we are doing other types of data structures too. Will that work?

Reply:
I don't think that makes sense, because the DSPlayground doesn't have access to the canvas, but the globalVars need access to the canvas in order to draw themselves.

The idea I had in mind is that there's actually nothing about the LinkedList class which is specific to linked lists (besides some of the initialization in the constructor). The only thing  specific to linked lists is the Node class, and the fact that it stores a single next pointer. Thus, I think we could potentially have an abstract Node class, and ListNode/TreeNode/etc. would inherit from that abstract class. Then, we could rename LinkedList to DataStructure, and the type of data structure would change when we change the type of Node that it stores. Does that sound reasonable?

Reply:
Ya that's a good point, abstracting it that way might work really nicely too. 

Lets just leave it as is then.